File: core/Core/Models/LoanPayment.cs
using System;
using System.Collections.Generic;

namespace Core.Models;

public partial class LoanPayment
{
    public int PaymentId { get; set; }

    public required int LoanId { get; set; }

    public required decimal Amount { get; set; }

    public required DateOnly PaymentDate { get; set; }

    public required bool IsPaid { get; set; }

    public required Loan Loan { get; set; }
}

File: analytic/domain/loan/loan_payment.py
from datetime import date
from decimal import Decimal


class LoanPayment:
    def __init__(self,
                 payment_id: int,
                 loan_id: int | None,
                 amount: Decimal | None,
                 payment_date: date | None,
                 is_paid: bool | None):
        self.payment_id = payment_id
        self.loan_id = loan_id
        self.amount = amount
        self.payment_date = payment_date
        self.is_paid = is_paid
